src/Sql/Sql/VirtualCluster/Cmdlet/RemoveAzureSqlVirtualCluster.cs

```diff
@@ -0,0 +1,145 @@
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Globalization;
+using System.Management.Automation;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.Sql.VirtualCluster.Model;
+
+namespace Microsoft.Azure.Commands.Sql.VirtualCluster.Cmdlet
+{
+    [Cmdlet(VerbsCommon.Remove, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "SqlVirtualCluster",
+        DefaultParameterSetName = RemoveByNameAndResourceGroupParameterSet,
+        SupportsShouldProcess = true),
+        OutputType(typeof(AzureSqlVirtualClusterModel))]
+    public class RemoveAzureSqlManagedInstance : VirtualClusterCmdletBase
+    {
+        protected const string RemoveByNameAndResourceGroupParameterSet =
+            "RemoveVirtualClusterFromInputParameters";
+
+        protected const string RemoveByInputObjectParameterSet =
+            "RemoveVirtualClusterFromAzureSqlVirtualClusterModelDefinition";
+
+        protected const string RemoveByResourceIdParameterSet =
+            "RemoveVirtualClusterFromAzureResourceId";
+
+        /// <summary>
+        /// Gets or sets the name of the Virtual Cluster.
+        /// </summary>
+        [Parameter(
+            ParameterSetName = RemoveByNameAndResourceGroupParameterSet,
+            Mandatory = true,
+            Position = 0,
+            HelpMessage = "The name of the virtual cluster.")]
+        [Alias("VirtualClusterName")]
+        [ResourceNameCompleter("Microsoft.Sql/virtualClusters", "ResourceGroupName")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the resource group to use.
+        /// </summary>
+        [Parameter(ParameterSetName = RemoveByNameAndResourceGroupParameterSet,
+            Mandatory = true,
+            Position = 1,
+            HelpMessage = "The name of the resource group.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public override string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Instance object to remove
+        /// </summary>
+        [Parameter(ParameterSetName = RemoveByInputObjectParameterSet,
+            Mandatory = true,
+            Position = 0,
+            ValueFromPipeline = true,
+            HelpMessage = "The instance object to remove")]
+        [ValidateNotNullOrEmpty]
+        [Alias("VirtualCluster")]
+        public AzureSqlVirtualClusterModel InputObject { get; set; }
+
+        /// <summary>
+        /// Gets or sets the resource id of the virtual cluster
+        /// </summary>
+        [Parameter(ParameterSetName = RemoveByResourceIdParameterSet,
+            Mandatory = true,
+            Position = 0,
```

Comment:
@milostod minor: you can remove this positional parameter since users will never be able to use it (if they provide a single string, PowerShell won't be able to differentiate between the Name/ResourceGroupName and ResourceId parameter sets, and since the Name/ResourceGroupName set is the default, it will choose that one and prompt the user to provide a value for `-ResourceGroupName`)

Reply:
I see, thanks. I've made the change.